File: src/cpp/chart.js
var max_width = 480;
var width = screen.width > max_width ? max_width: screen.width;

var chartCanvas = null;
var fillPercentage = 0;
var percentAdjustment = Math.PI/50;

function addRange(percentage, color)
{
    var context = chartCanvas.getContext("2d");
    context.fillStyle = color;

    context.beginPath();
    context.moveTo(width/2, width/2);

    var previousFill = fillPercentage*percentAdjustment;
    context.arc(width/2, width/2, width/2, previousFill, previousFill + (percentage*percentAdjustment));
    context.closePath();

    context.fill();

    fillPercentage += percentage;
}

function initialize()
{
    var chart = document.getElementById('chart');
    var canvas = document.createElement('canvas');

    canvas.width = width;
    canvas.height = width;

    chartCanvas = canvas;

    fillPercentage = 0;

    chart.appendChild(canvas);
}

function wipe()
{
    fillPercentage = 0;
    var context = chartCanvas.getContext("2d");
    context.clearRect(0, 0, chartCanvas.width, chartCanvas.height);  

    addRange(100, "gray");
    fillPercentage = 0;
}

initialize();
wipe();

File: src/csharp/Chart.cs
using System;
using System.Drawing;
using System.Drawing.Drawing2D;
using System.Drawing.Imaging;

public partial class Monitor
{
    private Int32 PreviousFill = 0;

    public void initialize()
    {
    }

    public void wipe()
    {
        PreviousFill = 0;
        Chart.Image = new Bitmap(CHART_SIZE,CHART_SIZE, PixelFormat.Format32bppPArgb);

        addRange(100, "gray");
        PreviousFill = 0;
    }

    public void addRange(Int32 Percentage, string Fill)
    {
        GraphicsPath Path = new GraphicsPath();

        Int32 Angle = Convert.ToInt32(Math.Ceiling(360.0*(Percentage/100.0)));

        const Int32 MID_POINT = (CHART_SIZE/2);
        Path.AddLine(MID_POINT, MID_POINT, MID_POINT, MID_POINT);
        Path.AddArc(0, 0, CHART_SIZE, CHART_SIZE, 270 + PreviousFill, Angle);
        PreviousFill += Angle;

        Graphics.FromImage(Chart.Image).FillPath(new SolidBrush(Color.FromName(Fill)), Path);
    }

}
